packages/next/src/client/app-index.tsx

```diff
@@ -225,8 +231,22 @@ export type ClientInstrumentationHooks = {
 }
 
 export function hydrate(
-  instrumentationHooks: ClientInstrumentationHooks | null
+  instrumentationHooks: ClientInstrumentationHooks | null,
+  assetPrefix: string
 ) {
+  let staticIndicatorState:
+    | { pathname: string | null; appIsrManifest: Record<string, true> }
+    | undefined
+  let webSocket: WebSocket | undefined
+
+  if (process.env.NODE_ENV !== 'production' && typeof window !== 'undefined') {
```

Comment:
The `typeof window` checks seems redundant. This is browser code and we're later in this function accessing `document` unconditionally.
```suggestion
  if (process.env.NODE_ENV !== 'production') {
```